lib/install/management/store_files.go

```diff
@@ -174,12 +172,10 @@ func (d *Dispatcher) deleteFilesIteratively(m *object.FileManager, ds *object.Da
 	return d.deleteVMFSFiles(m, ds, dsPath)
 }
 
-func (d *Dispatcher) deleteVMFSFiles(m *object.FileManager, ds *object.Datastore, dsPath string) error {
+func (d *Dispatcher) deleteVMFSFiles(m *object.DatastoreFileManager, ds *object.Datastore, dsPath string) error {
 	defer trace.End(trace.Begin(dsPath))
 
-	if _, err := tasks.WaitForResult(d.ctx, func(ctx context.Context) (tasks.Task, error) {
-		return m.DeleteDatastoreFile(ctx, dsPath, d.session.Datacenter)
-	}); err != nil {
+	if err := m.Delete(d.ctx, dsPath); err != nil {
```

Comment:
How are we handling NotAuthenticated issues from session timeout or similar? If there's any aspect style remediation performed in the tasks package beyond handling InProgress retrys then we should be using it across the board. Otherwise not.

@cgtexmex something else for vmomi gateway